diff --git a/core/modules/views/src/Plugin/views/query/QueryPluginBase.php b/core/modules/views/src/Plugin/views/query/QueryPluginBase.php
index 1e0740f..d6e9a60 100644
--- a/core/modules/views/src/Plugin/views/query/QueryPluginBase.php
+++ b/core/modules/views/src/Plugin/views/query/QueryPluginBase.php
@@ -149,7 +149,7 @@ public function setLimit($limit) {
    * Set an OFFSET on the query, specifying a number of results to skip.
    */
   public function setOffset($offset) {
-    $this->offset = $offset;
+    $this->offset = is_null($offset) ? 0 : $offset;
   }

   /**

